From 3d6d71c167dd9742cd072e7e6a3453c23ebb2658 Mon Sep 17 00:00:00 2001
From: "La Ode Muh. Fadlun Akbar" <fadlun.net@gmail.com>
Date: Thu, 12 Oct 2017 00:43:11 +0700
Subject: [PATCH] Update gvc submodule and fix issues when built using meson
 0.43.0

Because meson support was added to gvc, we can use subprojects in
in more elegant way. Also, meson will automatically clone the git
submodule to subprojects directory.
---
 .gitmodules                    |  2 +-
 meson.build                    |  7 +++++-
 src/applets/status/meson.build |  2 +-
 src/gvc                        |  1 -
 src/panel/meson.build          |  2 +-
 src/raven/meson.build          |  2 +-
 subprojects/gvc                |  1 +
 subprojects/gvc/meson.build    | 56 ------------------------------------------
 8 files changed, 11 insertions(+), 62 deletions(-)
 delete mode 160000 src/gvc
 create mode 160000 subprojects/gvc
 delete mode 100644 subprojects/gvc/meson.build

diff --git a/.gitmodules b/.gitmodules
index 4993d5d..f423fbb 100644
--- a/.gitmodules
+++ b/.gitmodules
@@ -1,3 +1,3 @@
 [submodule "gvc"]
-	path = src/gvc
+	path = subprojects/gvc
 	url = https://git.gnome.org/browse/libgnome-volume-control
diff --git a/meson.build b/meson.build
index d3441aa..be7352a 100644
--- a/meson.build
+++ b/meson.build
@@ -114,7 +114,12 @@ extra_includes = [
 with_polkit = get_option('with-polkit')
 
 # Get gvc built before we do anything
-gvc = subproject('gvc')
+gvc = subproject('gvc',
+    default_options: [
+    'c_std=c89',
+    'static=true',
+  ]
+)
 
 # Prep translations prior to build
 subdir('po')
diff --git a/src/applets/status/meson.build b/src/applets/status/meson.build
index 3db65f6..85a48b1 100644
--- a/src/applets/status/meson.build
+++ b/src/applets/status/meson.build
@@ -22,7 +22,7 @@ applet_status_deps = [
     dep_accountsservice,
     dependency('upower-glib', version: '>= 0.99.0'),
     link_libplugin,
-    gvc.get_variable('link_libgvc'),
+    gvc.get_variable('libgvc_dep'),
     meson.get_compiler('c').find_library('m', required: false),
 ]
 
diff --git a/src/gvc b/src/gvc
deleted file mode 160000
index ce8e488..0000000
--- a/src/gvc
+++ /dev/null
@@ -1 +0,0 @@
-Subproject commit ce8e4880ce31e275c40825c4ed756c791107f810
diff --git a/src/panel/meson.build b/src/panel/meson.build
index f6725a0..77c676b 100644
--- a/src/panel/meson.build
+++ b/src/panel/meson.build
@@ -50,7 +50,7 @@ panel_deps = [
     link_libtheme,
     link_libplugin,
     link_libraven,
-    gvc.get_variable('link_libgvc'),
+    gvc.get_variable('libgvc_dep'),
 ]
 
 top_vapidir = join_paths(meson.source_root(), 'vapi')
diff --git a/src/raven/meson.build b/src/raven/meson.build
index a6f268f..67248ae 100644
--- a/src/raven/meson.build
+++ b/src/raven/meson.build
@@ -35,7 +35,7 @@ libraven_deps = [
     link_libconfig,
     link_libplugin,
     link_libtheme,
-    gvc.get_variable('link_libgvc'),
+    gvc.get_variable('libgvc_dep'),
 ]
 
 top_vapidir = join_paths(meson.source_root(), 'vapi')
diff --git a/subprojects/gvc b/subprojects/gvc
new file mode 160000
index 0000000..8408721
--- /dev/null
+++ b/subprojects/gvc
@@ -0,0 +1 @@
+Subproject commit 84087213f1d06fdefb29d2a3eba8fc8c30c1ba93
diff --git a/subprojects/gvc/meson.build b/subprojects/gvc/meson.build
deleted file mode 100644
index 164553b..0000000
--- a/subprojects/gvc/meson.build
+++ /dev/null
@@ -1,56 +0,0 @@
-project(
-    'gvc', 'c',
-    version: '1.0',
-    license: [
-        'GPL-2.0',
-    ],
-    default_options: [
-        'c_std=c89',
-    ],
-)
-
-libgvc_deps = [
-    dependency('libpulse', version: '>= 2.0'),
-    dependency('libpulse-mainloop-glib', version: '>= 2.0'),
-]
-
-libgvc_sources = [
-    '../../src/gvc/gvc-channel-map.c',
-    '../../src/gvc/gvc-mixer-card.c',
-    '../../src/gvc/gvc-mixer-control.c',
-    '../../src/gvc/gvc-mixer-event-role.c',
-    '../../src/gvc/gvc-mixer-sink.c',
-    '../../src/gvc/gvc-mixer-sink-input.c',
-    '../../src/gvc/gvc-mixer-source.c',
-    '../../src/gvc/gvc-mixer-source-output.c',
-    '../../src/gvc/gvc-mixer-stream.c',
-    '../../src/gvc/gvc-mixer-ui-device.c',
-    '../../src/gvc/test-audio-device-selection.c',
-]
-
-libgvc = static_library(
-    'gvc',
-    libgvc_sources,
-    dependencies: libgvc_deps,
-    include_directories: [
-        include_directories('.'),
-        include_directories(join_paths('..', '..')),
-        include_directories(join_paths('..', '..', 'src', 'gvc')),
-    ],
-    c_args: [
-        '-DG_LOG_DOMAIN="Gvc"',
-        '-Wno-pedantic',
-        '-Wno-unused-parameter',
-        '-Wno-conversion',
-        '-Wno-sign-compare',
-    ],
-)
-
-link_libgvc = declare_dependency(
-    link_with: libgvc,
-    include_directories: [
-        include_directories('.'),
-        include_directories(join_paths('..', '..')),
-        include_directories(join_paths('..', '..', 'src', 'gvc')),
-    ],
-)
-- 
2.14.3

